Protocol error on arm64

I've compiled Criterion v2.3.3 from source on an ubuntu16 arm64 image without issue, but trying to run even the simplest test results in:

```
criterion: Re-entering criterion from a test worker. This is a catastrophic bug, please report it on the issue tracker.
Bailing out to avoid fork-bombing the system.
criterion: Could not spawn test instance: Protocol error
Aborted (core dumped)
```

Some version information:
```
ubuntu@ip-172-30-4-244:~/criterion$ ldd --version
ldd (Ubuntu GLIBC 2.23-0ubuntu11) 2.23

ubuntu@ip-172-30-4-244:~/criterion$ gcc --version
gcc (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609

ubuntu@ip-172-30-4-244:~/criterion$ cmake --version
cmake version 3.5.1

ubuntu@ip-172-30-4-244:~/criterion$ uname -a
Linux ip-172-30-4-244 4.15.0-1045-aws #47~16.04.1-Ubuntu SMP Fri Aug 2 16:23:47 UTC 2019 aarch64 aarch64 aarch64 GNU/Linux
```

Let me know if there's anything else I can do to help track down the issue!

I can't seem to get the latest commit of BoxFort to compile on that box:

```
ubuntu@ip-172-30-4-164:~/BoxFort$ mkdir build && cd build
ubuntu@ip-172-30-4-164:~/BoxFort/build$ cmake ..
-- The C compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Looking for _r_debug
-- Looking for _r_debug - found
-- Looking for _DYNAMIC
-- Looking for _DYNAMIC - found
-- Looking for PR_SET_PDEATHSIG
-- Looking for PR_SET_PDEATHSIG - found
-- Looking for shm_open in rt
-- Looking for shm_open in rt - found
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Looking for CLOCK_MONOTONIC_RAW
-- Looking for CLOCK_MONOTONIC_RAW - found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for environ
-- Looking for environ - found
-- Looking for mincore
-- Looking for mincore - found
-- Checking prototype mincore for HAVE_UNSIGNED_MINCORE_VEC - True
-- Looking for pthread_cond_timedwait_relative_np in pthread
-- Looking for pthread_cond_timedwait_relative_np in pthread - not found
-- Looking for trunc in m
-- Looking for trunc in m - found
-- Looking whether assembler supports `.hidden test`
-- Looking whether assembler supports `.hidden test` - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Configuring done
CMake Error at CMakeLists.txt:187 (add_library):
  Cannot find source file:

    src/asm/trampoline-arm64.S

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


CMake Error: CMake can not determine linker language for target: boxfort
CMake Error: Cannot determine link language for target "boxfort".
-- Generating done
-- Build files have been written to: /home/ubuntu/BoxFort/build
```

Looks like commit b926a6d broke cmake configuration for me, but even the commit before I can't actually compile:

```
ubuntu@ip-172-30-4-164:~/BoxFort/build$ cmake --build .
Scanning dependencies of target boxfort
[  5%] Building ASM object CMakeFiles/boxfort.dir/src/asm/trampoline-arm64.S.o
In file included from /home/ubuntu/BoxFort/src/asm/trampoline-arm64.S:25:0:
/home/ubuntu/BoxFort/src/asm/mangling.h:37:25: fatal error: mangling/BXF_MANGLING.h: No such file or directory
compilation terminated.
CMakeFiles/boxfort.dir/build.make:62: recipe for target 'CMakeFiles/boxfort.dir/src/asm/trampoline-arm64.S.o' failed
make[2]: *** [CMakeFiles/boxfort.dir/src/asm/trampoline-arm64.S.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/boxfort.dir/all' failed
make[1]: *** [CMakeFiles/boxfort.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

I do know that 926bd4c works though, because that's what Criterion builds as a dependency.

I'll give qemu a try following this guide : https://wiki.ubuntu.com/ARM64/QEMU

--- a/subprojects/boxfort.wrap
+++ b/subprojects/boxfort.wrap
@@ -1,4 +1,4 @@
 [wrap-git]
 directory = boxfort
 url = https://github.com/Snaipe/BoxFort.git
-revision = 38c49b14189fb66f0a97ea892d93ff4f948e9e00
+revision = cec0db7ab13c51d0baa01e2680b0edee2f7d317a

